### PR TITLE
Restructure post & jumpstart module, rename the post to article …

### PR DESCRIPTION
**Pull request by smebrahimi**
This issue #189
#### Rename post to article in the all project
1. store
2. components, views
3. router
4. text of hard code
5. filenames

#### Change flow add email & draft email
1. Step 1: Dropdown filled by dateTime of jumpstarts
2. Step 2: In the email draft show email preview by jumpStartId
3. Remove a condition of the slug for (Answer at a glance)
4. Change params of the body for the POST of jumpStart

### CODE REVIEW

**Comment by smebrahimi** [comment]
@ctyar PR is pending for backend's bugs

File: src/views/Email.vue [hunk 1]
```diff
@@ -82,11 +88,14 @@ export default {
       return true;
     },
     submit() {
-      let total_time = new Date(this.get_email_date + this.get_schedule_time).toISOString();
+      console.log(formatISODate(this.get_email_date));
+      const date = new Date(this.get_email_date.split('T')[0]).getTime();
```
Comment (mort3za) [inline src/views/Email.vue:92]: Seems this is unrelated and hacky. there should be a better way to convert that string to date.

**Comment by mort3za** [comment]
Please check all related functionalities before merge.

File: src/components/Email/SelectEmailDate.vue [hunk 2]
```diff
@@ -18,9 +23,20 @@
               </span>
             </template>
           </BaseDropdown>
-          <p v-if="$v.email_date.$error" class="text-xs text-error">
+          <p v-if="$v.email_date.$error && get_jumpstarts.length" class="text-xs text-error">
```
Comment (atharva3010) [inline src/components/Email/SelectEmailDate.vue:26]: What is the purpose of checking for `get_jumpstarts.length` here?

File: src/components/PageNavigation.vue [hunk 3]
```diff
@@ -35,10 +35,10 @@
       <div class="text-sm lg:flex-grow">
         <router-link
           v-if="isAuthenticated"
-          to="/posts/add"
+          to="/articles/add"
```
Comment (atharva3010) [inline src/components/PageNavigation.vue:38]: Can we replace this with `:to="{ name: 'ArticleAdd' }"`

File: src/views/Email.vue (hunk 1 above)
Comment (atharva3010) [reply src/views/Email.vue:92]: I think you want to do first `date.setHours(0, 0, 0, 0)` and then get milliseconds from that @smebrahimi. You can remove `.split('T')[0]`

File: src/components/Email/SelectEmailDate.vue [hunk 4]
```diff
@@ -43,45 +58,55 @@ export default {
...
-      let days = [];
-      for (let i = 0; i < 3; i++) {
-        let a = d + i * 86400000;
-        days.push(a);
-      }
-      return days;
+      if (!this.get_jumpstarts.length) return [];
+      const dates = [];
+      this.get_jumpstarts.forEach(jumpstart => {
+        dates.push(jumpstart.dateTime);
```
Comment (atharva3010) [inline src/components/Email/SelectEmailDate.vue:81]: We should convert these to timestamp/milliseconds, it's easier to manage time that way and also keeps things consistent.

File: src/components/Email/SelectEmailDate.vue [hunk 5]
```diff
@@ -43,45 +58,55 @@ export default {
...
+        this.fetch_jumpstarts();
+      }
+    },
+    prefillArticles(date) {
+      this.update_selected_articles();
+      this.fetch_jumpstarts().then(() => {
+        let i, article, selectedJumpstart;
+        const selectedDate = this.get_email_date;
+        selectedJumpstart = this.get_jumpstarts.find(item => {
+          return item.dateTime == selectedDate;
```
Comment (atharva3010) [inline src/components/Email/SelectEmailDate.vue:104]: You may need to convert `item.dateTime` into milliseconds too

File: src/components/Email/DraftEmail.vue [hunk 6]
```diff
@@ -3,34 +3,17 @@
...
         </span>
         <div class="grid grid-cols-1 md:grid-cols-2 gap-4 my-4">
-          <Post
-            v-for="post in get_posts"
-            :key="post.id"
-            :post="post"
-            @clicked="addPost(post)"
-            :active="isActive(post)"
+          <Article
+            v-for="article in get_selected_articles"
```
Comment (atharva3010) [inline src/components/Email/DraftEmail.vue:12]: shouldn't it be `get_articles` instead of `get_selected_articles`?
